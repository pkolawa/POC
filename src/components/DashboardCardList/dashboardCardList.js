import React from "react";
import { withStyles } from "@material-ui/core/styles";
import Container from "@material-ui/core/Container";
import Typography from "@material-ui/core/Typography";
import Paper from "@material-ui/core/Paper";
import SettingsIcon from "@material-ui/icons/Settings";
import TimelineIcon from "@material-ui/icons/Timeline";
import AutorenewIcon from "@material-ui/icons/Autorenew";
import { Link } from "react-router-dom";

const styles = (theme) => ({
  root: {
    marginTop: "30px",
    display: "flex",
    flexWrap: "wrap",
    justifyContent: "center",
    "& > *": {
      margin: theme.spacing(1),
      width: theme.spacing(24),
      height: theme.spacing(24),
    },
    "& a": {
      textDecoration: "none",
      display: "inline-flex",
      alignItems: "center",
      height: "inherit",
      width: "inherit",
      justifyContent: "center",
      flexDirection: "column",
      "& p": {
        fontWeight: "500"
      }
    },
  },
});

class Dashboard extends React.Component {
  render() {
    const { classes } = this.props;
    return (
      <React.Fragment>
        <Container>
          <div className={classes.root}>
            <Paper elevation={3}>
              <Link to="/app/configuration">
                <SettingsIcon color="primary" />
                <Typography align="center" color="primary">
                  Configuration Management
                </Typography>
              </Link>
            </Paper>
            <Paper elevation={3}>
              <Link to="/app/monitor">
                <TimelineIcon color="primary" />
                <Typography align="center" color="primary">
                  Monitor Page
                </Typography>
              </Link>
            </Paper>
            <Paper elevation={3}>
              <Link to="/app/runtime">
                <AutorenewIcon color="primary" />
                <Typography align="center" color="primary">
                  Runtime Management
                </Typography>
              </Link>
            </Paper>
          </div>
        </Container>
      </React.Fragment>
    );
  }
}

export {Dashboard}

export default withStyles(styles)(Dashboard);
