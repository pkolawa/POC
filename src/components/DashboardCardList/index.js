import Dashboard from "./dashboardCardList";

export { Dashboard as default };