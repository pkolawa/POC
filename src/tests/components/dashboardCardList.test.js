import React from "react";
import { shallow } from "enzyme";
import { getClasses } from "@material-ui/core/test-utils";
import { Dashboard } from "../../components/DashboardCardList/dashboardCardList";

test("Should Render Dashboard Card List", () => {
  const wrapper = shallow(<Dashboard classes={getClasses}/>);
  expect(wrapper).toMatchSnapshot();
});
