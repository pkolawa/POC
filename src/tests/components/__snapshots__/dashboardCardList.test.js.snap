// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Render Dashboard Card List 1`] = `
<Fragment>
  <WithStyles(ForwardRef(Container))>
    <div>
      <WithStyles(ForwardRef(Paper))
        elevation={3}
      >
        <Link
          to="/configuration"
        >
          <SettingsIcon
            color="primary"
          />
          <WithStyles(ForwardRef(Typography))
            align="center"
            color="primary"
          >
            Configuration Management
          </WithStyles(ForwardRef(Typography))>
        </Link>
      </WithStyles(ForwardRef(Paper))>
      <WithStyles(ForwardRef(Paper))
        elevation={3}
      >
        <Link
          to="/monitor"
        >
          <TimelineIcon
            color="primary"
          />
          <WithStyles(ForwardRef(Typography))
            align="center"
            color="primary"
          >
            Monitor Page
          </WithStyles(ForwardRef(Typography))>
        </Link>
      </WithStyles(ForwardRef(Paper))>
      <WithStyles(ForwardRef(Paper))
        elevation={3}
      >
        <Link
          to="/runtime"
        >
          <AutorenewIcon
            color="primary"
          />
          <WithStyles(ForwardRef(Typography))
            align="center"
            color="primary"
          >
            Runtime Management
          </WithStyles(ForwardRef(Typography))>
        </Link>
      </WithStyles(ForwardRef(Paper))>
    </div>
  </WithStyles(ForwardRef(Container))>
</Fragment>
`;
